priv/mysql.sql

```diff
@@ -58,7 +58,7 @@ CREATE INDEX i_last_seconds ON last(seconds);
 
 CREATE TABLE rosterusers (
     username varchar(250) NOT NULL,
-    jid varchar(250) NOT NULL,
```

Comment:
What does this comment mean? How does introducing prepared queries affect the schema?

Reply:
Basically, it's a contract on the data format with DB administrators.
For some reason, mod_roster assumes that there could be an invalid jids in DB, which never happens.
Unless, there is another app modifying the data.

So, it's a constraint that it's not just `NOT NULL`, but the data should be verified before inserting in the DB.
The current behaviour was to just ignore such data. Which is such a corner case actually and resulted in a lot of non-covered code.

Reply:
Does introducing prepared queries affect this or not?

Reply:
Not strictly. But each time we trying to parse value from `rosterusers.jid`, we have some logic attached to handle case when it's not a valid jid.
But this logic is unttested, uncovered with tests and I have severe questions about it (basically, it silently discards rows with invalid jids in some cases, but fails in another cases).

So, I've decided to remove "error handling logic" and just enforce the rule, that jids should be parsable. 

If you don't like comments in schema, I could remove them. But I think it could be helpful for some developers, who do not understand erlang.